rotkehlchen/chain/ethereum/modules/thegraph/accountant.py

```diff
@@ -0,0 +1,79 @@
+from rotkehlchen.chain.evm.accounting.structures import TxEventSettings
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.constants import ZERO
+from rotkehlchen.fval import FVal
+
+if TYPE_CHECKING:
+    from rotkehlchen.accounting.pot import AccountingPot
+    from rotkehlchen.accounting.structures.evm_event import EvmEvent
+    from rotkehlchen.assets.asset import Asset
+    from rotkehlchen.types import ChecksumEvmAddress
+
+
+class ThegraphAccountant(ModuleAccountantInterface):
+    def reset(self) -> None:
+        self.assets_supplied: dict[tuple[ChecksumEvmAddress, Asset], FVal] = defaultdict(FVal)
+
+    def _process_deposit(
+            self,
+            pot: 'AccountingPot',  # pylint: disable=unused-argument
+            event: 'EvmEvent',
+            other_events: Iterator['EvmEvent'],  # pylint: disable=unused-argument
+    ) -> None:
+        self.assets_supplied[(string_to_evm_address(event.location_label), event.asset)] += event.balance.amount  # type: ignore[arg-type]  # location_label can't be None here  # noqa: E501
+
+    def _process_withdraw(
+            self,
+            pot: 'AccountingPot',
+            event: 'EvmEvent',
+            other_events: Iterator['EvmEvent'],  # pylint: disable=unused-argument
+    ) -> None:
+        key = (string_to_evm_address(event.location_label), event.asset)  # type: ignore[arg-type]  # location_label can't be None here
+        self.assets_supplied[key] -= event.balance.amount
+        if self.assets_supplied[key] < ZERO:
+            gain = -1 * self.assets_supplied[key]
+            resolved_asset = event.asset.resolve_to_asset_with_symbol()
+            pot.add_acquisition(
+                event_type=AccountingEventType.TRANSACTION_EVENT,
+                notes=f'Gained {gain} {resolved_asset.symbol} as delegation reward for {event.location_label}',  # noqa: E501
+                location=event.location,
+                timestamp=event.get_timestamp_in_sec(),
+                asset=event.asset,
+                amount=gain,
+                taxable=True,
+                extra_data={'tx_hash': event.tx_hash.hex()},
+            )
+            self.assets_supplied[key] = ZERO
+
+    def event_settings(self, pot: 'AccountingPot') -> dict[str, TxEventSettings]:  # pylint: disable=unused-argument
+        return {
+            get_tx_event_type_identifier(HistoryEventType.STAKING, HistoryEventSubType.DEPOSIT_ASSET, CPT_THEGRAPH): TxEventSettings(  # noqa: E501
+                taxable=False,
+                count_entire_amount_spend=False,
+                count_cost_basis_pnl=False,
+                method='spend',
+                accountant_cb=self._process_deposit,
+            ),
+            get_tx_event_type_identifier(HistoryEventType.SPEND, HistoryEventSubType.FEE, CPT_THEGRAPH): TxEventSettings(  # noqa: E501
+                taxable=False,
+                count_entire_amount_spend=False,
+                count_cost_basis_pnl=False,
```

Comment:
This is why the test discrepancy. Also I know this may be a big difference but perhaps you should add a generic rule here: https://github.com/rotki/rotki/blob/dec19d6aa1b5ef325b6bd0d913504d448d1fe6b1/rotkehlchen/accounting/history_base_entries.py#L35

Perhaps the generic rule can come as a surgical PR later, replacing this and any other generic fees that are all counted the same, to see if it makes sense. Problem is we don't have too many accounting tests (compared to other parts of the app) so just CI passing won't tell us everything.

```suggestion
                taxable=True,
                count_entire_amount_spend=True,
                count_cost_basis_pnl=True,
```

Reply:
New place after my PR. Please rebase it 😓 
 https://github.com/rotki/rotki/blob/06f0dfcecb6a34be1cb6266420cd40ad8021c783/rotkehlchen/accounting/rules.py#L23